src/main/java/Parser.java

```diff
@@ -0,0 +1,89 @@
+     * If user input consist of keyword list, a List object will be returned.
+     * IF user input consist of keyword bye, a Exit object will be returned.
+     * else a DukeException will be thrown with invalid command tagged to it.
+     *
+     * @param fullCommand String object of input given by user.
+     * @return Command object that represents the command of the user input.
+     * @throws DukeException             if user input is empty, invalid or Command description is empty.
+     * @throws DateTimeException         if Date format is not in yyyy-mm-dd.
+     * @throws IndexOutOfBoundsException if index accessing is more than amount of Task exist or negative index.
+     */
+    public static Command parse(String fullCommand) throws DukeException, DateTimeException, IndexOutOfBoundsException {
+        String[] inputArr = fullCommand.split(" ");
+        String command = inputArr[0];
+        if (fullCommand.length() == 0) {
+            throw new DukeException("Type something", DukeErrorType.EMPTY_COMMAND);
+        }
+        CommandList commandValue;
+        try {
+            commandValue = CommandList.valueOf(command.toUpperCase());
+        } catch (IllegalArgumentException exception) {
+            throw new DukeException("Invalid command", DukeErrorType.INVALID_COMMAND);
+        }
+        switch (commandValue) {
+        case TODO:
+            String[] todoArr = fullCommand.split("todo");
+            if (todoArr.length == 0 || todoArr[1].trim().length() == 0) {
+                throw new DukeException("Empty ToDo description", DukeErrorType.EMPTY_DESCRIPTION, command);
+            } else {
+                String todoDescription = todoArr[1].trim();
+                return new Add(new ToDo(todoDescription));
+            }
+        case EVENT:
+            String[] eventDetails = fullCommand.split("/at");
+            String eventTime = eventDetails[1].trim();
+            String[] descriptionArr = eventDetails[0].split("event");
+            String eventDescription = descriptionArr[1].trim();
+            if (eventDetails.length == 0) {
+                throw new DukeException("Empty Event time", DukeErrorType.EMPTY_TIME, command);
+            } else if (descriptionArr.length == 0 || eventDescription.length() == 0) {
+                throw new DukeException("Empty Event description", DukeErrorType.EMPTY_DESCRIPTION, command);
+            } else {
+                return new Add(new Event(eventDescription, LocalDate.parse(eventTime, USER_FORMAT)));
+            }
+        case DEADLINE:
+            String[] deadlineDetails = fullCommand.split("/by");
+            String deadlineTime = deadlineDetails[1].trim();
+            String deadlineDescription = "";
+            String[] descriptionArrDeadLine = deadlineDetails[0].split("deadline");
+            deadlineDescription = descriptionArrDeadLine[1].trim();
+            return new Add(new Deadline(deadlineDescription, LocalDate.parse(deadlineTime, USER_FORMAT)));
```

Comment:
Perhaps you could add spaces in front of your round brackets '(' to improve readability of your code.

Reply:
I've changed this line to 
``` 
return new Add(new Deadline(deadlineDescription, LocalDate.parse(deadlineTime, USER_FORMAT)));`
```
``` 
               return new Add(
                        new Deadline(deadlineDescription, LocalDate.parse(deadlineTime, USER_FORMAT)));
```
Is it more readable? Or should is this what u meant?:

```
return new Add( new Deadline(deadlineDescription, LocalDate.parse(deadlineTime, USER_FORMAT)));
```